app/scripts/controllers/paymentModal.js

```diff
@@ -143,16 +141,45 @@ angular.module('confRegistrationWebApp')
       }
       $scope.paymentToRefund = payment;
 
+      var refundAmount;
+
+      if($scope.isPartialRefundAvailable(payment, payment.paymentType)) {
+        refundAmount = $scope.calculateRefundableAmount(payment);
+        $scope.onlyFullRefund = false;
+      }
+      else {
+        refundAmount = payment.amount;
+        $scope.onlyFullRefund = true;
+      }
+
       $scope.refund = {
-        amount: $scope.calculateRefundableAmount(payment),
+        amount: refundAmount,
         refundedPaymentId: payment.id,
         registrationId: payment.registrationId,
         paymentType: 'REFUND',
-        refundChannel: payment.paymentType,
-        readyToProcess: true
+        refundChannel: payment.paymentType
       };
     };
 
+    $scope.refreshRefund = function (refund) {
```

Comment:
It determines whether a partial refund is available, and if not, disables the amount window and shows the warning.